pkg/common/utils.go

```diff
@@ -115,6 +123,56 @@ func ConvertSITimeout(millis int64) time.Duration {
 	return time.Duration(result)
 }
 
+// ConvertSITimeoutWithAdjustment Similar to ConvertSITimeout, but this function also adjusts the timeout if
+// "creationTime" is defined. It's used during Yunikorn restart, in order to properly track how long a placeholder pod should
+// be in "Running" state.
+func ConvertSITimeoutWithAdjustment(siApp *si.AddApplicationRequest) time.Duration {
+	result := ConvertSITimeout(siApp.ExecutionTimeoutMilliSeconds)
```

Comment:
What happens if I do not have a timeout set? Setting the default is part of the application code which means we'll break at this point. Setting the default should be moved into the `ConvertSITimeout()` call.

Reply:
Fixed.